packages/webdriverio/src/session/context.ts

```diff
@@ -65,13 +65,19 @@ export class ContextManager extends SessionManager {
         }
     }
 
-    #onCommandResultBidiAndClassic(event: { command: string, result: unknown }) {
+    async #onCommandResultBidiAndClassic(event: { command: string, result: unknown }) {
         /**
          * the `closeWindow` command returns:
          *   > the result of running the remote end steps for the Get Window Handles command, with session, URL variables and parameters.
          */
         if (event.command === 'closeWindow') {
-            this.#currentContext = (event.result as { value: string[] }).value[0]
+            const windowHandles = (event.result as { value: string[] }).value
+            if (windowHandles.length === 0) {
+                const newWindowHandle = await this.#browser.createWindow('tab')
+                this.#currentContext = newWindowHandle.handle
+                return this.#browser.switchToWindow(this.#currentContext)
```

Comment:
How about instead of calling a command that will fail anyway to just throw a meaningful message here, saying that all handles were removed which has caused WebdriverIO to close the session.

Reply:
You mean instead of creating a new window, we just return an error message? That would not resolve this issue, although that's what the [spec](https://www.w3.org/TR/webdriver2/#dfn-close-window:~:text=If%20there%20are%20no%20more%20open%20top%2Dlevel%20browsing%20contexts%2C%20then%20try%20to%20close%20the%20session.) tells to do.

> calling a command that will fail anyway

Which command are we referring to here?